src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/AssociatedMetadataProvider.cs

```diff
@@ -163,16 +166,21 @@ private TModelMetadata CreateTypeInformation(Type type, IEnumerable<Attribute> a
                 attributes = attributes.Concat(associatedAttributes);
             }
 
+            attributes = modelMetadataAttrHelper.GetModelMetadataAttributesOnType(type, attributes);
+
             return CreateMetadataPrototype(attributes, containerType: null, modelType: type, propertyName: null);
         }
 
         private PropertyInformation CreatePropertyInformation(Type containerType, PropertyHelper helper)
         {
             var property = helper.Property;
+            var attributes = ModelAttributes.GetAttributesForProperty(property);
+            attributes = modelMetadataAttrHelper.GetModelMetadataAttributesOnProperty(containerType, attributes, property.Name);
```

Comment:
how will this work (and ensure correct attribute ordering) when we fix #1000 and are checking for attributes on the property's `Type` as well?